File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 0;
    
    const list = clothes.reduce((list,product)=>{
        list[product[1]] = (list[product[1]] || 0) +1;
        return list
    },{});
    
    let values = Object.values(list)
    answer = values.reduce((total, options) => total * (options + 1), 1) - 1;
    
    
    
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    //전체 갯수 + 얼굴 x 상의 x 하의 x 겉옷
    // 그냥 딕셔너리에 key:부위 value:갯수 하면 될듯
    int answer = 1;
    
    unordered_map<string, int> cnt;
    cnt.reserve(clothes.size());
    
    for (const vector<string>& name : clothes) {
        cnt[name[1]]++;
    }
    
    for (const auto& [key, value] : cnt) {
        answer += answer * value;
    }
    
    
    return answer-1;
}